apps/verify.c

```diff
@@ -246,12 +246,21 @@ static int check(X509_STORE *ctx, const char *file,
 
     if (sm2id != NULL) {
 #ifndef OPENSSL_NO_SM2
-        ASN1_OCTET_STRING v;
+        ASN1_OCTET_STRING *v;
 
-        v.data = sm2id;
-        v.length = sm2idlen;
+        v = ASN1_OCTET_STRING_new();
+        if (v == NULL) {
+            printf("error: SM2 ID allocation failed\n");
```

Comment:
Shouldn't this be BIO_printf to bio_err?

Reply:
I just followed other `printf`s in this function, necessary to change to BIO_printf?

Reply:
We appear to be inconsistent in this file - we use both BIO_printf and printf here. I think we should always be using BIO_printf.

Reply:
I get the point.